packages/react-aria-components/test/Menu.test.tsx

```diff
@@ -1056,4 +1050,259 @@ describe('Menu', () => {
       expect(menu).not.toBeInTheDocument();
     });
   });
+
+  describe('portalContainer', () => {
+    function InfoMenu(props) {
+      return (
+        <UNSTABLE_PortalProvider getContainer={() => props.container.current}>
+          <MenuTrigger>
+            <Button aria-label="trigger" />
+            <Popover>
+              <Menu aria-label="Test">
+                <MenuItem id="1">One</MenuItem>
+                <MenuItem id="">Two</MenuItem>
+                <MenuItem id="3">Three</MenuItem>
+              </Menu>
+            </Popover>
+          </MenuTrigger>
+        </UNSTABLE_PortalProvider>
+      );
+    }
+
+    function App() {
+      let container = React.useRef(null);
+      return (
+        <>
+          <InfoMenu container={container} />
+          <div ref={container} data-testid="custom-container" />
+        </>
+      );
+    }
+
+    it('should render the menu in the portal container', async () => {
+      let {getByRole, getByTestId} = render(
+        <App />
+      );
+
+      let button = getByRole('button');
+      await user.click(button);
+
+      expect(getByRole('menu').closest('[data-testid="custom-container"]')).toBe(getByTestId('custom-container'));
+    });
+  });
+});
+
+// better to accept items from the test? or just have the test have a requirement that you render a certain-ish structure?
+// what about the button label?
+// where and how can i define the requirements/assumptions for setup for the test?
```

Comment:
good question, the first thing that comes into mind is those ascii art like comments that some libraries for visualization lol. Not sure tbh, will think more on this

Reply:
haha, yeah, i thought about that, but was too lazy to make an ascii art, sounds hard

Reply:
yup, we can decide what we wanna do about this once we settle on the structure for these tests